include/vinecopulib/vinecop/implementation/class.ipp

```diff
@@ -1070,7 +1072,7 @@ Vinecop::rosenblatt(const Eigen::MatrixXd& u, const size_t num_threads) const
         }
 
         // h-functions are only evaluated if needed in next step
-        Bicop edge_copula = get_pair_copula(tree, edge).as_continuous();
+        Bicop edge_copula = pair_copulas_[tree][edge].as_continuous();
```

Comment:
Shouldn't it also be a pointer here?

Reply:
As it is now, `as_continuous()` always makes exactly one copy. If variables are already continuous, the pair copula is just returned as a copy. If it is discrete, a temporary copy is created inside the function and then automatically moved into `edge_copula` (see [Return Value Optimization](https://en.wikipedia.org/wiki/Copy_elision#:~:text=In%20the%20context%20of%20the,a%20function's%20return%20value). So a pointer wouldn't save us any memory here.